### PR TITLE
fix: check the diskName

### PR DESCRIPTION
**Pull request by ZeroMagic**
**What type of PR is this?**
/kind bug
/kind failing-test

**What this PR does / why we need it**:
Check the diskName before creating azure disk.

**Which issue(s) this PR fixes**:

Fixes #46 

**Special notes for your reviewer**:

**Release note**:
```
fix: set disk name as volume name
```

### CODE REVIEW

File: pkg/azuredisk/azuredisk.go [hunk 1]
```diff
@@ -315,3 +322,34 @@ func (d *Driver) CreateManagedDisk(ctx context.Context, options *ManagedDiskOpti
...
+
+// The name of the managed disk that is being created.
+// he name can't be changed after the disk is created.
+// Supported characters for the name are a-z, A-Z, 0-9 and _.
+// The maximum name length is 80 characters.
+//
+// See https://docs.microsoft.com/en-us/rest/api/compute/disks/createorupdate#uri-parameters
+func getValidDiskName(volumeName string) string {
+	diskName := strings.Replace(volumeName, "-", "_", -1)
+	if len(diskName) > diskNameMaxLength {
```
Comment (feiskyer) [inline pkg/azuredisk/azuredisk.go]: add a warning log here since the name is truncated.

@andyzhangx any possible of same names after truncation?

Comment (ZeroMagic) [reply pkg/azuredisk/azuredisk.go]: Done. 

In the next step, we will call azure cloud's api to create volume. In the api, the name will be checked if it exists.

Comment (andyzhangx) [reply pkg/azuredisk/azuredisk.go]: looks like the disk name could contain `-` , while according to doc, it said: 
```
 Supported characters for the name are a-z, A-Z, 0-9 and _. The maximum name length is 80 characters.
```
Let me ask azure disk team about this doc issue.

Comment (andyzhangx) [reply pkg/azuredisk/azuredisk.go]: @ZeroMagic Let's use the naming rule azure portal:
- _ . are all allowed chars

File: pkg/azuredisk/azuredisk.go [hunk 2]
```diff
@@ -315,3 +322,36 @@ func (d *Driver) CreateManagedDisk(ctx context.Context, options *ManagedDiskOpti
...
+	}
+
+	return diskName
+}
+
+func checkDiskName(diskName string) bool {
+	length := len(diskName)
+
+	for i, v := range diskName {
+		if !(unicode.IsLetter(v) || unicode.IsDigit(v) || v == '_' || v == '.' || v == '-') ||
```
Comment (andyzhangx) [inline pkg/azuredisk/azuredisk.go]: `- _ .` are all allowed chars

Comment (andyzhangx) [reply pkg/azuredisk/azuredisk.go]: fix this

File: pkg/azuredisk/azuredisk_test.go [hunk 3]
```diff
@@ -200,3 +200,109 @@ func TestIsValidDiskURI(t *testing.T) {
...
+		{
+			diskName: "1_",
+			expected: true,
+		},
+		{
+			diskName: "_1",
+			expected: false,
+		},
+		{
+			diskName: "1.",
```
Comment (andyzhangx) [inline pkg/azuredisk/azuredisk_test.go]: add some positive cases, e.g. "1.2", "1_2", "1-3"

File: pkg/azuredisk/azuredisk.go (hunk 2 above)
Comment (ZeroMagic) [reply pkg/azuredisk/azuredisk.go]: done

File: pkg/azuredisk/azuredisk_test.go (hunk 3 above)
Comment (ZeroMagic) [reply pkg/azuredisk/azuredisk_test.go]: done

**Pull request merged**